python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -1425,6 +1429,268 @@ keys:
+            dot1q:
+              type: dict
+              keys:
+                ethertype:
+                  type: int
+                  convert_types:
+                    - str
+                  min: 1536
+                  max: 65535
+                  description: Ethertype/TPID (Tag Protocol IDentifier) for VLAN tagged frames.
+                vlan_tag:
+                  type: str
+                  description: Allow/disallow VLAN tagged frames.
+                  valid_values:
+                    - disallowed
+                    - required
+            source_interface:
+              type: str
+              description: |-
+                tx: Allow bridged packets to go out of the source interface.
+                tx multicast: Allow multicast packets only to go out of the source interface.
+              valid_values:
+                - tx
+                - tx multicast
+            access_vlan:
+              type: int
+              convert_types:
+                - "str"
+              min: 1
+              max: 4094
+              description: |-
+                Set VLAN when interface is in access mode.
+                Warning: This should not be combined with `ethernet_interfaces[].mode = access/dot1q-tunnel`.
+            phone:
+              type: dict
+              keys:
+                vlan:
+                  type: int
+                  min: 1
+                  max: 4094
+                  convert_types:
+                    - str
+                  description: |-
+                    Warning: This should not be combined with `ethernet_interfaces[].phone.vlan`.
+                trunk:
+                  type: str
+                  description: |-
+                    Warning: This should not be combined with `ethernet_interfaces[].phone.trunk`
+                  valid_values: ["tagged", "tagged phone", "untagged", "untagged phone"]
+            mode:
+              type: str
+              valid_values: ["access", "dot1q-tunnel", "trunk", "trunk phone"]
+              description: |-
+                Warning: This should not be combined with `ethernet_interfaces[].mode`
+            pvlan_mapping:
+              type: str
+              convert_types:
+                - int
+              description: |-
+                Secondary VLAN IDs of the private VLAN mapping.
+                Warning: This should not be combined with `ethernet_interfaces[].pvlan_mapping`.
+            vlan_translations:
+              description: |-
+                Warning: This should not be combined with `ethernet_interfaces[].vlan_translations`.
+              type: dict
+              keys:
+                direction_in:
+                  type: list
+                  description: Map ingress packets only.
+                  items:
+                    type: dict
+                    keys:
+                      required:
```

Comment:
May be keep this required out of the direction_list because it is configured once.

Reply:
As discussed, keeping the key name same as CLI.
we are defining separate sections for in, out and both. So keeping the `required` key inside `direction_in`

Reply:
So according this schema we can add multiple required keys in direction_in.

direction_in:
  - required: true
  -  from: 100
      to: 105
  - required: false

This creates confusion. 
So I think we can add:

vlan_translation:
  required: true
  direction_in:
  direction_out:

{% if ethernet_interface.switchport.vlan_translation.required is arista.avd.defined(true) %}
  switchport vlan transation in required
{% endif %}

Reply:
the required is only for `direction_in` here

Reply:
got your point here, moved the required key.
thanks!